application/src/main/java/com/sanction/thunder/authentication/KeyAuthenticator.java

```diff
@@ -1,24 +1,25 @@
 package com.sanction.thunder.authentication;
 
 import com.google.common.base.Optional;
-
-import io.dropwizard.auth.AuthenticationException;
 import io.dropwizard.auth.Authenticator;
-
 import io.dropwizard.auth.basic.BasicCredentials;
 
 import java.util.List;
 
-public class ThunderAuthenticator implements Authenticator<BasicCredentials, Key> {
-
+public class KeyAuthenticator implements Authenticator<BasicCredentials, Key> {
   private final List<Key> allKeys;
 
-  public ThunderAuthenticator(List<Key> keys) {
+  public KeyAuthenticator(List<Key> keys) {
     this.allKeys = keys;
   }
 
-  @Override
-  public Optional<Key> authenticate(BasicCredentials credentials) throws AuthenticationException {
+  /**
+   * Checks the incoming key against config keys.
+   *
+   * @param credentials basic auth credentials
+   * @return ThunderPrincipal
+   */
+  public Optional<Key> authenticate(BasicCredentials credentials) {
```

Comment:
I think you can keep an `@Override` annotation on this method. Then you won't need a Javadoc comment